node/libs/roles/src/validator/messages/leader_commit.rs

```diff
@@ -76,18 +96,28 @@ impl CommitQC {
 
     /// Add a validator's signature.
     /// Signature is assumed to be already verified.
-    pub fn add(&mut self, msg: &Signed<ReplicaCommit>, genesis: &Genesis) {
+    pub fn add(
+        &mut self,
+        msg: &Signed<ReplicaCommit>,
+        genesis: &Genesis,
+    ) -> Result<(), CommitQCAddError> {
+        use CommitQCAddError as Error;
         if self.message != msg.msg {
-            return;
+            return Err(Error::InconsistentViews);
```

Comment:
The error is not quite right. If the message differs it can be because the view is different or because the proposal is different. Better to rename it.

Reply:
Ah! you are right. Fixing it